mlir-tensorrt/compiler/lib/Dialect/Plan/Transforms/CreateClosedRegions.cpp

```diff
@@ -351,70 +351,11 @@ static void remapAnalysisState(DataFlowSolver &solver, ValueRange originals,
   }
 }
 
-static LogicalResult createClosedGroupOp(RewriterBase &rewriter,
-                                         plan::InlineGroupOp op,
-                                         DataFlowSolver &solver) {
-  OpBuilder::InsertionGuard g(rewriter);
-  Location loc = op.getLoc();
-
-  // Materialize the destination operands.
-  SmallVector<DestinationOperandMaterializationResult> destinationOperands;
-  destinationOperands.reserve(op.getNumResults());
-  for (OpResult res : op->getOpResults()) {
-    FailureOr<DestinationOperandMaterializationResult> destResult =
-        materializeDestinationOperand(rewriter, res.getLoc(), op,
-                                      res.getResultNumber(), solver);
-    if (failed(destResult))
-      return emitError(res.getLoc())
-             << "failed to materialize destination operand of type "
-             << res.getType();
-    destinationOperands.push_back(*destResult);
-  }
-
-  // Make the region isolated from above. This captures the input operands.
-  SmallVector<Value> inputs = makeRegionIsolatedFromAbove(
-      rewriter, op.getRegion(), [&](Operation *producer) {
-        return shouldCloneProducer(producer, op.getRegion());
-      });
-
-  rewriter.setInsertionPoint(op);
-  auto closedGroupOp = rewriter.create<InlineClosedGroupOp>(
-      op.getLoc(), /*target=*/op.getTarget(),
-      /*inputs=*/inputs,
-      /*outs=*/
-      llvm::map_to_vector(destinationOperands,
-                          [](const auto &x) { return x.destinationOperand; }));
-
-  rewriter.inlineBlockBefore(
-      &op.getRegion().front(), &closedGroupOp.getRegion().front(),
-      closedGroupOp.getRegion().front().end(),
-      closedGroupOp.getRegion().getArguments().take_front(
-          op.getRegion().getNumArguments()));
-
-  SmallVector<Value> replacements;
-  replacements.reserve(destinationOperands.size());
-  for (auto [newResult, destOperand, originalType] :
-       llvm::zip(closedGroupOp->getResults(), destinationOperands,
-                 op->getResultTypes())) {
-    if (destOperand.strategy ==
-        DestinationOperandMaterializationStrategy::ExactShape) {
-      replacements.push_back(newResult);
-      continue;
-    }
-
-    assert(destOperand.exactShape &&
-           "expected materialized shape values to be provided for "
-           "this materialization strategy");
-
-    replacements.push_back(newResult);
-  }
-
-  // Since we are about to replace values that may be inputs to other regions
-  // ops, we need to update the solver to populate the replacement TensorKind
-  // information.
-  remapAnalysisState(solver, op->getResults(), replacements);
-  rewriter.replaceOp(op, replacements);
+static FailureOr<SmallVector<BoundsAttr>>
+getInputAttributes(RewriterBase &rewriter, DataFlowSolver &solver, Location loc,
+                   SmallVector<Value> const &inputs) {
```

Comment:
```suggestion
                  ValueRange inputs) {
```

In general, use `ValueRange` and `TypeRange` to represent const range over values and types, respectively. For other cases, use `ArrayRef<T>` rather than `const SmallVector<T> &`. For mutable SmallVector, use `SmallVectorImpl<T> &` to allow for different implicit size parameters. See note here https://llvm.org/docs/ProgrammersManual.html#llvm-adt-smallvector-h.

Reply:
Thanks a lot! This is super helpful.